### ISSUE TITLE
Next.js + Apollo Server 3.0 + Subscription 

### ISSUE THREAD

**Issue by nenjamin2405** [apollographql/apollo-server#5568]
Hi guys, I successfully integrated Apollo Server 3.0 into Next.js route called /api by using below code snippet

```javascript
import { ApolloServer } from "apollo-server-micro"; // NOSONAR
import { typeDefs } from "../../graphql/schema";
import { resolvers } from "../../graphql/resolvers";

import Cors from "micro-cors";
import { API_URL } from "../../constants/common";

const cors = Cors();

const apolloServer = new ApolloServer({
  resolvers,
  typeDefs,
  introspection: true,
});

// @ts-ignore
await apolloServer.start();

export default cors(async (req, res) => {
  if (req.method === "OPTIONS") {
    res.end();
    return false;
  }

  return apolloServer.createHandler({ path: API_URL })(req, res);
});

export const config = {
  api: {
    bodyParser: false,
  },
};

```

Right now I want to integrate the Subscription functionality into this api, but I cannot get it working. 

This stackoverflow [topic](https://stackoverflow.com/questions/62371284/apollo-server-with-subscriptions-used-inside-next-js-api-routes-websockets-trou) is outdated, and the apollo-server-micro also removed the integrated subscription in the Apollo Server instance. 

If I tried to follow the example from Apollo [here](https://www.apollographql.com/docs/apollo-server/data/subscriptions/), The last step requires me to run a httpServer by following code:

```javascript
(async function () {
  const app = express();

  const httpServer = createServer(app);

  const schema = makeExecutableSchema({
    typeDefs,
    resolvers,
  });

  const server = new ApolloServer({
    schema,
  });
  await server.start();
  server.applyMiddleware({ app });

  SubscriptionServer.create(
    { schema, execute, subscribe },
    { server: httpServer, path: server.graphqlPath }
  );

  const PORT = 4000;
  httpServer.listen(PORT, () =>
    console.log(`Server is now running on http://localhost:${PORT}/graphql`)
  );
})();
```

Looking at the code snippet above, I have no idea how to convert it into Next.js's requestHandler.

In conclusion, do you guys have any suggestion of how to enable Subscription Server in Next.js route using Apollo Server 3.0, because we already got this working in the Apollo Server 2.0.

Many thanks!

**Comment by mehdyben** [comment]
I got the same issue, integrating Apollo Server 3.0 into Next.js is now a pain.

**Comment by glasser** [comment]
Apollo Server 3 doesn't have built-in subscription support, and `subscriptions-transport-ws` (ie `SubscriptionServer`) is currently effectively unmaintained.

Did this even work in Apollo Server 2? Do you have links to examples of using Apollo Server 2 with subscriptions and next.js? Might be able to help you adapt that. I do see this https://github.com/vercel/next.js/issues/9829 so I'm not so sure...

**Comment by HamadaBrest** [comment]
@nenjamin2405 

I've successfully integrated `Apollo server 3` with `Next.js` using `Relay` as a client this way : 

✨ Server side : 
```js
(async function () {
  const app = express();

  const httpServer = createServer(app);

  const schema = makeExecutableSchema({
    typeDefs,
    resolvers,
  });

  const server = new ApolloServer({
    schema,
  });
  await server.start();
  server.applyMiddleware({ app });

  SubscriptionServer.create(
    { schema, execute, subscribe },
    { server: httpServer, path: ['SUBSCRIPTION_PATH'] }
  );

  const PORT = 4000;
  httpServer.listen(PORT, () =>
    console.log(`Server is now running on http://localhost:${PORT}/graphql`)
  );
})();
```

✨ Client side : 

```js
import {SubscriptionClient} from 'subscriptions-transport-ws';
import {Observable} from 'relay-runtime';

const subscriptionClient = new SubscriptionClient(['SUBSCRIPTION_PATH'], {
    reconnect: true,
});

return Observable.create(sink =>
    subscriptionClient.request({query: config.text, variables}).subscribe(sink)
);
```

NB 🔔 : In `Apollo server v2`, Subscriptions were built-in, nos with the `v3`, you need a standalone server with a unique `port` for subscriptions.
